### ISSUE TITLE
The request url is outside of Vite serving allow list.

### ISSUE THREAD

**Issue by adriancalavie** [sveltejs/kit#10427]
### Describe the bug

After initializing a new SvelteKit Skeleton, project, and running `npm run dev -- --open`, if you open the site, you'll get the following message:

`The request url "path/to/project/node_modules/@sveltejs/kit/src/runtime/client/start.js" is outside of Vite serving allow list.`

Every time you click on a hyperlink you get this message again. 

Also it seems to block certain SvelteKit featues like preload data.

### Reproduction

- create a new SvelteKit Project using `npm create svelte@latest test-project`. Select Skeleton, Typescript and no options.
- `cd test-project`
- `npm install`
- `npm run dev -- --open`
- open the browser to http://localhost:5173/
- The message should be logged in the console

Additional testing:
- create an 'about' page
- create a hyperlink to it
- by default, SvelteKit has preload-data set to "hover", so try to hover over the hyperlink.
- no requests should be done on the hover (You can check using the Network Tab)

### Logs

```Shell
PS C:\Users\adrian\Dev\web> npm create svelte@latest test-project

create-svelte version 5.0.3

┌  Welcome to SvelteKit!
│
◇  Which Svelte app template?
│  Skeleton project
│
◇  Add type checking with TypeScript?
│  Yes, using TypeScript syntax
│
◇  Select additional options (use arrow keys/space bar)
│  none
│
└  Your project is ready!

✔ Typescript
  Inside Svelte components, use <script lang="ts">

Install community-maintained integrations:
  https://github.com/svelte-add/svelte-add

Next steps:
  1: cd test-project
  2: npm install (or pnpm install, etc)
  3: git init && git add -A && git commit -m "Initial commit" (optional)
  4: npm run dev -- --open

To close the dev server, hit Ctrl-C

Stuck? Visit us at https://svelte.dev/chat
PS C:\Users\adrian\Dev\web> cd .\test-project\
PS C:\Users\adrian\Dev\web\test-project> npm install

added 111 packages, and audited 112 packages in 13s

11 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
PS C:\Users\adrian\Dev\web\test-project> npm run dev -- --open

> test-project@0.0.1 dev
> vite dev


Forced re-optimization of dependencies

  VITE v4.4.6  ready in 662 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h to show help
The request url "D:/Dev/web/test-project/node_modules/@sveltejs/kit/src/runtime/client/start.js" is outside of Vite serving allow list.

- C:/Users/adrian/Dev/web/test-project/src/lib
- C:/Users/adrian/Dev/web/test-project/src/routes
- C:/Users/adrian/Dev/web/test-project/.svelte-kit
- C:/Users/adrian/Dev/web/test-project/src
- C:/Users/adrian/Dev/web/test-project/node_modules
- D:/Dev/web/test-project/node_modules/vite/dist/client

Refer to docs https://vitejs.dev/config/server-options.html#server-fs-allow for configurations and more details.
```

### System Info

```Shell
System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700KF
    Memory: 17.44 GB / 31.84 GB
  Binaries:
    Node: 20.4.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.8.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1992.0), Chromium (115.0.1901.183)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @sveltejs/adapter-auto: ^2.0.0 => 2.1.0
    @sveltejs/kit: ^1.20.4 => 1.22.3
    svelte: ^4.0.5 => 4.1.1
    vite: ^4.4.2 => 4.4.6
```

### Severity

annoyance

**Comment by s3812497** [comment]
What does the `--` argument input do for the vite dev command? Shouldn't it just be `npm run dev --open`?
I suspect the `--` argument is being interpreted as the vite dev `root` argument and causing the issue.
https://vitejs.dev/guide/cli.html

**Comment by adriancalavie** [comment]
AFAIK, the `--` argument is used to signal the end of npm-specific commands and the beginning of custom script arguments. Anything after this will be passed directly to the script defined in "dev".

I've tried it without the double hyphen, and the outcome is the same.

**Comment by adriancalavie** [comment]
Update: 

This issue was caused by a symbolic link I had that ... linked `C:/Dev` to `D:/Dev`.
Because I ran `npm run dev -- --open` from ` C:/Users/adrian/Dev/web/test-project/` and basically had a symbolic link at the middle of my path, Vite was searching for some files in `C:/Users/adrian/Dev/web/test-project/node_modules` while the files were actually in `D:/Dev/web/test-project/node_modules/vite/dist/client`. 

So, it was my mistake after all, not knowing how symbolic links affect tools like this.

**Comment by betaglot** [comment]
What did you change to fix this?

**Comment by adriancalavie** [comment]
It was a long time ago, but I think I scrapped the sym link alltogether, and since then I've kept my dev projects inside `C:/Users/adrian/Developer`, with nothing on other partitions (like `D:/`). 

So yeah, if you're in the same boad as I was, I recommend moving the project in a set location and avoid sym links as they can cause problems like this. 

Also, there's now the option of [Dev drives](https://learn.microsoft.com/en-us/windows/dev-drive/). While they're targeting performance gains, one could also use this to organize their filesystem and keep dev things inside a dev things folder/drive. 

Good luck!
